app/forms/fields/decimal_field_with_separator.py

```diff
@@ -15,13 +16,25 @@ class DecimalFieldWithSeparator(DecimalField):
     DecimalPlace validators
     """
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
+    def __init__(
+        self,
+        *,
+        description: str,
+        label: str | None = None,
+        validators: Sequence[Callable],
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(
+            description=description,
+            label=label,
+            validators=validators,
+            **kwargs,
+        )
         self.data = None
 
-    def process_formdata(self, valuelist):
+    def process_formdata(self, valuelist: Sequence[str] | None = None) -> None:
         if valuelist:
             try:
-                self.data = Decimal(sanitise_number(valuelist[0]))
+                self.data = Decimal(sanitise_number(valuelist[0]))  # type: ignore # None type is safeguarded in if statement above
```

Comment:
same as my other comment about the type ignore, can be fixed in the init with `self.data: int | None = None`